misc/systemd/ipfs.service

```diff
@@ -0,0 +1,31 @@
+[Unit]
+description=ipfs p2p daemon
+After=network.target
+Requires=network.target
+
+[Service]
+Type=simple
+User=ipfs
+RestartSec=1
+Restart=always
+PermissionsStartOnly=true
+Nice=18
+StateDirectory=/var/lib/ipfs
+Environment=IPFS_PATH=/var/lib/ipfs
+Environment=HOME=/var/lib/ipfs
+LimitNOFILE=8192
+Environment=IPFS_FD_MAX=8192
+EnvironmentFile=-/etc/sysconfig/ipfs
+StandardOutput=journal
+WorkingDirectory=/var/lib/ipfs
+ExecStartPre=-adduser --system --group --home /var/lib/ipfs ipfs
+ExecStartPre=-mkdir /var/lib/ipfs
+ExecStartPre=-/bin/chown ipfs:ipfs /var/lib/ipfs
+ExecStartPre=-/bin/chmod u+rwx /var/lib/ipfs
+ExecStartPre=-chpst -u ipfs /usr/local/bin/ipfs init --profile=badgerds
+ExecStartPre=-chpst -u ipfs /usr/local/bin/ipfs config profile apply server
```

Comment:
This being a systemwide service file assumes that the "user" is root and that there is one ipfs daemon running on the system - just like ssh, or httpd, or any other system service.  They don't need to be run each time but otherwise there are additional steps to installing/starting ipfs and it doesn't hurt anything to run them each startup.

Reply:
> They don't need to be run each time but otherwise there are additional steps to installing/starting ipfs and it doesn't hurt anything to run them each startup.

They *will* run each time the service is started or restarted, which also means that the configuration as specified in $IPFS_PATH/config will be overwritten each time. Sure it is great for when you are installing ipfs the first time but is annoying everytime after that.

Reply:
I have not found this systemd unit file to be annoying in several weeks of use, and if an administrator that uses the example file does, it is trivial to remove any lines that they find to be cumbersome or unnecessary.